src/IECoreScene/CurvesAlgo.cpp

```diff
@@ -334,146 +336,23 @@ struct  CurvesVaryingToVertex
-// todo: note it's a duplicate of DeleteFlaggedFaceVaryingFunctor in MeshAlgo.cpp
-template<typename U>
-class DeleteFlaggedVertexFunctor
-{
-	public:
-		typedef DataPtr ReturnType;
-
-		DeleteFlaggedVertexFunctor(  typename IECore::TypedData<std::vector<U> >::ConstPtr flagData, ConstIntVectorDataPtr verticesPerCurve, bool invert ) : m_flagData( flagData ), m_verticesPerCurve( verticesPerCurve ), m_invert ( invert )
-		{
-		}
-
-		template<typename T>
-		ReturnType operator()( const T *data )
-		{
-			const typename T::ValueType &inputs = data->readable();
-			const std::vector<int> &verticesPerCurve= m_verticesPerCurve->readable();
-			const std::vector<U> &flags = m_flagData->readable();
-
-			T *filteredResultData = new T();
-			typename T::ValueType &filteredResult = filteredResultData->writable();
-
-			filteredResult.reserve( inputs.size() );
-
-			size_t offset = 0;
-			for( size_t c = 0; c < verticesPerCurve.size(); ++c )
-			{
-				int numVerts = verticesPerCurve[c];
-				if( (m_invert && flags[c]) || (!m_invert && !flags[c]) )
-				{
-					for( int v = 0; v < numVerts; ++v )
-					{
-						filteredResult.push_back( inputs[offset + v] );
-					}
-				}
-				offset += numVerts;
-			}
-
-			return filteredResultData;
-		}
-	private:
-		typename IECore::TypedData<std::vector<U> >::ConstPtr m_flagData;
-		ConstIntVectorDataPtr m_verticesPerCurve;
-		bool m_invert;
-};
-
-template<typename U>
-class DeleteFlaggedVaryingFunctor
-{
-	public:
-		typedef DataPtr ReturnType;
-
-		DeleteFlaggedVaryingFunctor( typename IECore::TypedData<std::vector<U> >::ConstPtr flagData, const CurvesPrimitive* curvesPrimitive, bool invert ) : m_flagData( flagData ), m_curvesPrimitive (curvesPrimitive), m_invert ( invert )
-		{
-		}
-
-		template<typename T>
-		ReturnType operator()( const T *data )
-		{
-			const typename T::ValueType &inputs = data->readable();
-			const std::vector<U> &flags = m_flagData->readable();
-
-			T *filteredResultData = new T();
-			typename T::ValueType &filteredResult = filteredResultData->writable();
-
-			filteredResult.reserve( inputs.size() );
-
-			size_t offset = 0;
-			for( size_t c = 0; c < m_curvesPrimitive->numCurves(); ++c )
-			{
-				int numVarying = m_curvesPrimitive->numSegments( c ) + 1;
-
-				if( (m_invert && flags[c]) || (!m_invert && !flags[c]) )
-				{
-					for( int v = 0; v < numVarying; ++v )
-					{
-						filteredResult.push_back( inputs[offset + v] );
-					}
-				}
-				offset += numVarying;
-			}
-
-			return filteredResultData;
-		}
-	private:
-		typename IECore::TypedData<std::vector<U> >::ConstPtr m_flagData;
-		const CurvesPrimitive *m_curvesPrimitive;
-		bool m_invert;
-};
 
 template<typename T>
-CurvesPrimitivePtr deleteCurves( const CurvesPrimitive *curvesPrimitive, const typename IECore::TypedData<std::vector<T> > *deleteFlagData, bool invert )
+CurvesPrimitivePtr deleteCurves(
+	const CurvesPrimitive *curvesPrimitive,
+	PrimitiveVariable::IndexedView<T>& indexedView,
```

Comment:
`indexedView` doesn't tell us what this variable does - the previous name was more informative.